lib/split/user.rb

```diff
@@ -38,28 +48,41 @@ def cleanup_old_versions!(experiment)
 
     def active_experiments
       experiment_pairs = {}
-      user.keys.each do |key|
-        Metric.possible_experiments(key_without_version(key)).each do |experiment|
+      keys_without_finished(user.keys).each do |key|
+        Metric.possible_experiments(Split::Experiment.key_without_version_for_experiment_key(key)).each do |experiment|
           if !experiment.has_winner?
-            experiment_pairs[key_without_version(key)] = user[key]
+            experiment_pairs[Split::Experiment.key_without_version_for_experiment_key(key)] = user[key]
           end
         end
       end
       experiment_pairs
     end
 
+    def key_for_experiment(experiment)
+      keys.find { |k| k.match(Regexp.new("^#{experiment.name}"))}
+    end
+
+    def version_for_experiment(experiment)
+      kfe = key_for_experiment(experiment)
+      if kfe
+        kfe.split(":").last
+      else
+        0
+      end
```

Comment:
Doesn't the experiment have a version attribute we can grab?

Reply:
Yeah, don't know what happened here

Reply:
Ok I see now. It's really that we need the version of the experiment that concerns the current user. The change muddies this concept a little; the experiment itself does not have the full information.

Reply:
If we were both initially confused about this, it's probably going to be even more confusing six months after we've thought about it. Worth thinking about how to define this functionality such that this explanation is clear. It makes sense now that it's been explained fwiw, but I don't know that I would have figured this out as it is.

Reply:
Agreed. For lack of a better option. I will try to document in comments to give as much possible clarity.

Reply:
ok, addressed, at least minimally.